internal/ent/notifications/events.go

```diff
@@ -0,0 +1,399 @@
+			ownerID = policy.OwnerID
+		}
+		if approverID == "" && policy.ApproverID != "" {
+			approverID = policy.ApproverID
+		}
+	}
+
+	return approverID, name, entityID, ownerID, nil
+}
+
+func addTaskAssigneeNotification(ctx *soiree.EventContext, assigneeID, taskTitle, taskID, ownerID string) error {
+	client, ok := soiree.ClientAs[*generated.Client](ctx)
+	if !ok {
+		return nil
+	}
+
+	consoleURL := client.EntConfig.Notifications.ConsoleURL
+
+	// create the data map with the URL
+	dataMap := map[string]string{
+		"url": fmt.Sprintf("%s/tasks?id=%s", consoleURL, taskID),
+	}
+
+	dataJSON, err := json.Marshal(dataMap)
+	if err != nil {
+		logx.FromContext(ctx.Context()).Error().Err(err).Msg("failed to marshal notification data")
+		return err
+	}
+
+	data := notificationData{
+		notificationType: enums.NotificationTypeUser.String(),
+		userIDs:          []string{assigneeID},
+		title:            "New task assigned",
+		body:             fmt.Sprintf("Task %s has been assigned to you", taskTitle),
+		data:             string(dataJSON),
+		ownerID:          ownerID,
+		topic:            "task_assignment",
+		objectType:       "Task",
+	}
+
+	return newNotificationCreation(ctx, data)
+}
+
+func addInternalPolicyNotification(ctx *soiree.EventContext, approverID, policyName, policyID, ownerID string) error {
+	client, ok := soiree.ClientAs[*generated.Client](ctx)
+	if !ok {
+		return nil
+	}
+
+	// set allow context to query the group
+	allowCtx := privacy.DecisionContext(ctx.Context(), privacy.Allow)
+
+	users, err :=
+		client.GroupMembership.Query().Where(groupmembership.GroupID(approverID)).All(allowCtx)
+	if err != nil {
+		logx.FromContext(ctx.Context()).Error().Err(err).Str("group_id", approverID).Msg("failed to get approver group")
+		return err
+	}
+
+	if len(users) == 0 {
+		logx.FromContext(ctx.Context()).Warn().Str("group_id", approverID).Msg("no users found in approver group")
+		return nil
+	}
+
+	// collect user IDs
+	userIDs := make([]string, len(users))
+	for i, user := range users {
+		userIDs[i] = user.ID
+	}
+
+	consoleURL := client.EntConfig.Notifications.ConsoleURL
+
+	// create the data map with the URL
+	dataMap := map[string]string{
+		"url": fmt.Sprintf("%s/policies/%s/view", consoleURL, policyID),
+	}
+
+	dataJSON, err := json.Marshal(dataMap)
+	if err != nil {
+		logx.FromContext(ctx.Context()).Error().Err(err).Msg("failed to marshal notification data")
+		return err
+	}
+
+	data := notificationData{
+		notificationType: enums.NotificationTypeOrganization.String(),
+		userIDs:          userIDs,
+		title:            "Policy approval required",
+		body:             fmt.Sprintf("%s needs approval, internalPolicy", policyName),
+		data:             string(dataJSON),
+		ownerID:          ownerID,
+		topic:            "policy_approval",
+		objectType:       "InternalPolicy",
+	}
+
+	return newNotificationCreation(ctx, data)
+}
+
+func newNotificationCreation(ctx *soiree.EventContext, data notificationData) error {
```

Comment:
We can use this instead:

```suggestion
func newNotificationCreation(ctx *soiree.EventContext, data *generated.CreateNotificationInput) error {
```